File: frontend/src/src/actions/admin_permission.ts
import { gql } from "@apollo/client";

export const GET_ADMIN_PERMISSIONS = gql`
  query GetAdminPermissions(
    $filters: [Filter!]
    $orders: [Order!]
    $p: Pagination
  ) {
    response: adminPermissions(
      filters: $filters
      orders: $orders
      pagination: $p
    ) {
      adminPermissions {
        id
        ip
        game
        membership
        statistical
        financials
        userId
        qna
        settlement
        dwdelete
        sale
        status
        user {
          id
          userid
        }
        createdAt
        updatedAt
        deletedAt
      }
      total
    }
  }
`;

export const FILTER_ADMIN_PERMISSIONS = gql`
  query FilterAdminPermissions(
    $filters: [Filter!]
    $orders: [Order!]
    $pagination: Pagination
  ) {
    response: filterAdminPermissions(
      filters: $filters
      orders: $orders
      pagination: $pagination
    ) {
      users {
        id
        name
        userid
        role
        rootId
        root {
          id
          userid
        }
        parentId
        parent {
          id
          userid
        }
        type
        blackMemo
        usdtAddress
        currentIP
        IP
        createdAt
        updatedAt
        status
        profile {
          id
          userId
          name
          nickname
          bankName
          holderName
          accountNumber
          birthday
          phone
          mobile
          balance
          point
          comp
          level
          favorites
          referral
          coupon
          lastDeposit
          lastWithdraw
        }
      }
      total
    }
  }
`;

export const UPDATE_ADMIN_PERMISSION = gql`
  mutation UpdateAdminPermission(
    $id: ID!
    $input: UpdateAdminPermissionInput!
  ) {
    success: updateAdminPermission(id: $id, input: $input) {
      id
      userId
    }
  }
`;

File: backend/src/internal/models/admin_permission.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type AdminPermission struct {
	ID uint `json:"id"`

	UserID uint `json:"userId" gorm:"unique;not null"`
	User   User `json:"user" gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`

	Membership  bool `json:"membership"`
	Financials  bool `json:"financials"`
	Qna         bool `json:"qna"`
	Game        bool `json:"game"`
	Settlement  bool `json:"settlement"`
	Sale        bool `json:"sale"`
	Statistical bool `json:"statistical"`
	IP          bool `json:"ip"`
	Dwdelete    bool `json:"dwdelete"`

	Status bool `json:"status"`

	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
	DeletedAt *gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}
